rules/msk_unique_app_names.go

```diff
@@ -0,0 +1,123 @@
+package rules
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/terraform-linters/tflint-plugin-sdk/hclext"
+	"github.com/terraform-linters/tflint-plugin-sdk/logger"
+	"github.com/terraform-linters/tflint-plugin-sdk/tflint"
+)
+
+const commonNameAttribute = "cert_common_name"
+
+type MSKUniqueAppNamesRule struct {
+	tflint.DefaultRule
+}
+
+func (r *MSKUniqueAppNamesRule) Name() string {
+	return "msk_unique_app_names"
+}
+
+func (r *MSKUniqueAppNamesRule) Enabled() bool {
+	return true
+}
+
+func (r *MSKUniqueAppNamesRule) Link() string {
+	return ReferenceLink(r.Name())
+}
+
+func (r *MSKUniqueAppNamesRule) Severity() tflint.Severity {
+	return tflint.ERROR
+}
+
+func (r *MSKUniqueAppNamesRule) Check(runner tflint.Runner) error {
+	isRoot, err := isRootModule(runner)
+	if err != nil {
+		return err
+	}
+	if !isRoot {
+		logger.Debug("skipping child module")
+		return nil
+	}
+
+	TLSAppModules, err := getTLSAppModules(runner)
+	if err != nil {
+		return err
+	}
+
+	return r.reportDuplicateTLSAppNames(runner, TLSAppModules)
+}
+
+func getTLSAppModules(runner tflint.Runner) (hclext.Blocks, error) {
+	modules, err := runner.GetModuleContent(
+		&hclext.BodySchema{
+			Blocks: []hclext.BlockSchema{
+				{
+					Type:       "module",
+					LabelNames: []string{"name"},
+					Body: &hclext.BodySchema{
+						Attributes: []hclext.AttributeSchema{
+							{Name: commonNameAttribute},
+						},
+					},
+				},
+			},
+		},
+		nil,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("getting modules: %w", err)
+	}
+
+	var TLSAppModules hclext.Blocks
+	for _, moduleBlock := range modules.Blocks {
+		if _, ok := moduleBlock.Body.Attributes[commonNameAttribute]; ok {
```

Comment:
This is a good point to only select the modules with the cert_common_name.

Wondering if it wouldn't be better to look at the source attribute and to get the ones with the tls-app suffix.

Also, we should do this for the msk_app_topics too.

Reply:
On another thought I understand why you took only the ones with cert_common_name. It is a required attribute and it is validated by the tflint validate plugin. 

So I think we should actually do both in the end.. filter only for modules with tls-app source and take only the ones with cert_common_name.

Reply:
> This is a good point to only select the modules with the cert_common_name.
> 
> Wondering if it wouldn't be better to look at the source attribute and to get the ones with the tls-app suffix.
> 
> Also, we should do this for the msk_app_topics too.

My approach has just been to filter based on the existence of the key I'm linting (e.g. `{consume,produce}_topic` in the `msk_app_topics` case) what's the value in additionally checking on the `tls-app` suffix? Unless we think we'll run into naming conflicts?

Reply:
I'm thinking that maybe in the future we'll have more than one module with the cert_common_name (maybe the quota module?) and in this way we'll make the targeting specific.
We can also leave it like this for now and handle it later, but I think the change should be quite easy to implement

Reply:
>  We can also leave it like this for now and handle it later, but I think the change should be quite easy to implement

I'm going to advocate for implementing this later: if/when we need it